pkg/v1/remote/index.go

```diff
@@ -122,42 +117,6 @@ func (r *remoteIndex) ImageIndex(h v1.Hash) (v1.ImageIndex, error) {
 	return desc.ImageIndex()
 }
 
-func (r *remoteIndex) imageByPlatform(platform v1.Platform) (v1.Image, error) {
-	desc, err := r.childByPlatform(platform)
-	if err != nil {
-		return nil, err
-	}
-
-	// Descriptor.Image will handle coercing nested indexes into an Image.
-	return desc.Image()
-}
-
-// This naively matches the first manifest with matching platform attributes.
-//
-// We should probably use this instead:
-//	 github.com/containerd/containerd/platforms
-//
-// But first we'd need to migrate to:
-//   github.com/opencontainers/image-spec/specs-go/v1
-func (r *remoteIndex) childByPlatform(platform v1.Platform) (*Descriptor, error) {
-	index, err := r.IndexManifest()
-	if err != nil {
-		return nil, err
-	}
-	for _, childDesc := range index.Manifests {
-		// If platform is missing from child descriptor, assume it's amd64/linux.
-		p := defaultPlatform
-		if childDesc.Platform != nil {
-			p = *childDesc.Platform
-		}
-
-		if matchesPlatform(p, platform) {
-			return r.childDescriptor(childDesc, platform)
-		}
-	}
-	return nil, fmt.Errorf("no child with platform %s/%s in index %s", platform.OS, platform.Architecture, r.Ref)
```

Comment:
Let's have partial return a similar error.

Reply:
And maybe fix up some things to include the ref in the equivalent error from remote? I'm not sure how useful it is to include.